cmake/external_rules.cmake

```diff
@@ -88,7 +91,7 @@ function(download_external_sources)
   endif()
 
   execute_process(
-    COMMAND ${ENV_COMMAND} cmake --build . -- ${cmake_build_args}
+    COMMAND ${ENV_COMMAND} ${CMAKE_COMMAND} --build . -- ${cmake_build_args}
```

Comment:
Should there really be a space between `--` and `${cmake_build_args}` ?

Reply:
I'm just not clear on why we use the "--" at all - typically that's how you separate sets of arguments, for example the arguments afterwards are sent to a subprocess, or are considered a list of filenames even if they start with a "-". But why is that necessary here?

(For example, if you have a filename called "-q", you can't normally run "grep hello -q" because grep will interpret "-q" as an argument. So you must do "grep hello -- -q" which tells grep that everything following the "--" is a filename. But I'm still not sure why that's required for this cmake call.)

Reply:
The more you know.

Reply:
Good catch. I didn't spot this and am not sure why this exists. I substituted hard coded cmake to using a variable.

Reply:
What's with the "--" though?

Reply:
The only change that I made is "cmake" converted to "${CMAKE_COMMAND}". Will investigate why the "-- ${cmake_build_args}" exists in another PR as I don't want to introduce new changes for this Windows/Android task.